### DISCUSSION TITLE
Clings/skids

### DISCUSSION THREAD

**Discussion by TF3RDL** [ekiefl/pooltool#196] [Ideas]
AFAIK, the friction for these balls are responsible for [throws](https://billiards.colostate.edu/tutorial/throw/) and [clings/skids](https://billiards.colostate.edu/faq/throw/cling/) are caused when a high-friction spot (where it is a chalk mark) is happened to be at contact point.

However, a paper about [Bounce Maps](https://graphics.stanford.edu/projects/bouncemap/) makes me curious because the idea of spatially-varying physical properties are nothing new, can something like this be applied for the friction instead of elasticity and these maps are dynamically changing (every time a cue strikes the ball, it imparts high-friction spot where there is a chalk mark)?

**Comment by ekiefl** [comment]
This is cool, thanks for sharing.

I think it's definitely possible, and I'll definitely keep it in mind for the future. Although at the current moment, not even _uniform_ ball friction is implemented. You wouldn't be interested in becoming a contributor, would you?

**Comment by ekiefl** [comment]
Moved this to Discussions.

Also, small update: uniform friction ball-ball friction is now implemented in two models:

https://pooltool.readthedocs.io/en/latest/autoapi/pooltool/physics/resolve/ball_ball/index.html#pooltool.physics.resolve.ball_ball.FrictionalInelastic
https://pooltool.readthedocs.io/en/latest/autoapi/pooltool/physics/resolve/ball_ball/index.html#pooltool.physics.resolve.ball_ball.FrictionalMathavan
